client/src/components/network-diagram/index.tsx

```diff
@@ -0,0 +1,121 @@
+'use client';
+import React, { useState } from 'react';
+
+import { CollapsibleContent, Collapsible } from '@radix-ui/react-collapsible';
+
+import { Organization, Project } from '@/types/generated/strapi.schemas';
+
+import { useNetworkDiagram } from '@/hooks/networks';
+
+import Item from './item';
+
+const NetworkDiagram = ({
+  data,
+  id,
+  type,
+}: {
+  data: Project | Organization;
+  id: number;
+  type: 'project' | 'organization';
+}) => {
+  const { name } = data;
+
+  const [openCollapsibles, setOpenCollapsibles] = useState<number[]>([]);
+  const {
+    data: networks,
+    isError,
+    isFetched,
+  } = useNetworkDiagram({
+    type,
+    id,
+  });
+  if (isFetched && isError) {
+    return null;
+  }
+  const getIndex = (childIndex: number) => {
+    const openedChildrenBeforeCurrent = networks.filter(
+      (n, nIndex) => n?.id !== undefined && openCollapsibles?.includes(n.id) && nIndex < childIndex,
+    );
+
+    const childIndexOffset = openedChildrenBeforeCurrent.reduce((acc, n) => {
+      const childrenLength = n?.children?.length || 0;
+      const isFirstNetwork = networks[0]?.id === n?.id;
+      return acc + childrenLength - 1 + (isFirstNetwork ? 1 : 0);
+    }, 0);
+
+    return childIndex + childIndexOffset;
+  };
+
+  return (
+    <div>
+      <div className="my-6 border-t border-dashed border-t-gray-200" />
+      <div className="mb-6 font-serif text-2xl text-slate-700">Network</div>
+      <div className="flex justify-between">
+        <div>
+          <Item name={name} id={id} type={type} hasChildren={networks?.length > 0} />
+          {networks.map((network, childIndex) => {
+            if (
+              !network ||
+              typeof network === 'undefined' ||
+              typeof network.name === 'undefined' ||
+              typeof network.id === 'undefined'
+            ) {
+              return null;
+            }
+
+            return (
+              <Collapsible key={network.id} className="ml-10">
+                <Item
+                  key={network.id}
+                  name={network.name}
+                  type={network.type as 'organization' | 'project'}
+                  category={network.category}
+                  id={network.id}
+                  setOpenCollapsibles={setOpenCollapsibles}
+                  openCollapsibles={openCollapsibles}
```

Comment:
I don't think this is a good idea because you're coupling the two components. Why not something like this?
```suggestion
                  open={openCollapsibles.includes(id)}
                  onToggle={(open) => setOpenCollapsibles(collapsibles => open ? [...collapsibles, id] : collapsibles.filter(collapsibleId => collapsibleId !== id))}
```

You could even extract the `onToggle` callback.